htdocs/societe/list.php

```diff
@@ -97,7 +97,7 @@
 $search_idprof5 = trim(GETPOST('search_idprof5', 'alpha'));
 $search_idprof6 = trim(GETPOST('search_idprof6', 'alpha'));
 $search_vat = trim(GETPOST('search_vat', 'alpha'));
-$search_sale = GETPOSTINT("search_sale");
+$search_sale = GETPOST("search_sale", 'alphanohtml');
```

Comment:
OK for the first suggestion 
if (GETPOSTISARRAY('search_sale')) {
	$search_sale = GETPOST('search_sale', 'array:int');
} else if (GETPOSTISSET('search_sale')) {
	$search_sale = array(GETPOSTINT('search_sale'));
}
for the second 
$search_sale_req  must be different de $search_sale 
because it must not contain "-2"  = (no sales staff)

the case of -2 
```
elseif (count($search_sale) > 0 && in_array('-2', $search_sale)) {
		$sql .= " AND (EXISTS (SELECT sc.fk_soc FROM ".MAIN_DB_PREFIX."societe_commerciaux as sc WHERE sc.fk_soc = s.rowid AND sc.fk_user IN (".$db->sanitize($search_sale_req)."))";
		$sql .= " OR NOT EXISTS (SELECT sc.fk_soc FROM ".MAIN_DB_PREFIX."societe_commerciaux as sc WHERE sc.fk_soc = s.rowid))";
	}
```
in my block following 
$search_sale_req = implode(',', $search_sale_req);
I need both variables

Reply:
@eldy